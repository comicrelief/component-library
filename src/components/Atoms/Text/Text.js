import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

/**
 * Text component
 */
export const BaseText = styled.span`
  color: ${({ color, theme }) =>
    color ? theme.color(color) : 'inherit'};
  font-size: ${({ size, theme }) => theme.fontSize(size)};
`;

/** Text renders different elements based on the `tag` prop */
const Text = ({ tag, children, ...rest }) => (
  <BaseText {...rest} as={tag}>
    {children}
  </BaseText>
);

Text.defaultProps = {
  tag: 'span',
  weight: 'normal',
  uppercase: false,
  size: 's0'
};

Text.propTypes = {
  /** Weight of Font */
  weight: PropTypes.string,
  /** Sets text transform to uppercase. */
  uppercase: PropTypes.bool,
  /** Colors */
  color: PropTypes.string,
  /** Tag type */
  tag: PropTypes.string,
  /** Sizes */
  size: PropTypes.string
};

export default Text;
