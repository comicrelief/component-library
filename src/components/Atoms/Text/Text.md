```js
<div>
  <Text tag="h3" size="s">
    Heading 3
  </Text>
  <Text tag="p" colour="black">
    My paragraph
  </Text>
  <Text tag="h3" size="m">
    Heading 3 size 3
  </Text>
  <Text tag="p" size="xl" color="red">
    My paragraph small and yellow
  </Text>
</div>
```

Sport Relief

```js
  <div>
    <Text tag="h3" size="s">
      Heading 3
    </Text>
    <Text tag="p" colour="black">
      My paragraph
    </Text>
    <Text tag="h3" size="m">
      Heading 3 size 3
    </Text>
    <Text tag="p" size="xl" color="red">
      My paragraph small and yellow
    </Text>
  </div>
```
