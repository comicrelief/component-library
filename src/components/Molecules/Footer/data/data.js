export default {
  menuDescription: 'Footer',
  menuGroup: [
    {
      id: 'f7dc5eef-f4eb-5405-8a35-08808b55cb33',
      title: 'Hear from us',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Get the newsletter',
          url: 'https://www.comicrelief.com/#',
          class: null
        },
        {
          title: 'How we contact you',
          url: 'https://www.comicrelief.com/update-your-preferences',
          class: null
        }
      ]
    },
    {
      id: '9f3980f3-d02b-52e9-ac41-933a778c040a',
      title: 'Get in touch',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Contact us',
          url: 'https://www.comicrelief.com/contact-us',
          class: null
        },
        {
          title: 'Your Gift Aid',
          url: 'https://giftaid.comicrelief.com/update',
          class: null
        },
        {
          title: 'FAQs',
          url: 'https://www.comicrelief.com/frequently-asked-questions',
          class: null
        },
        {
          title: 'Red Nose Day 2019 FAQs',
          url:
            'https://www.comicrelief.com/red-nose-days-frequently-asked-questions',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998b',
      title: 'About us',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Mission',
          url: 'https://www.comicrelief.com/about-comic-relief',
          class: null
        },
        {
          title: 'Our history',
          url: 'https://www.comicrelief.com/about-comic-relief/history',
          class: null
        },
        {
          title: 'Finances',
          url: 'https://www.comicrelief.com/about-comic-relief/finances',
          class: null
        },
        {
          title: 'Partners',
          url: 'https://www.comicrelief.com/working-with-us',
          class: null
        },
        {
          title: 'Meet the team',
          url: 'https://www.comicrelief.com/about-comic-relief/meet-the-team',
          class: null
        },
        {
          title: 'Sport Relief',
          url: 'https://www.sportrelief.com/',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373c',
      title: 'Careers',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Working at Comic Relief',
          url: 'https://www.comicrelief.com/careers',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://app.beapplied.com/org/comic-relief',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373d',
      title: 'News',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'News',
          url: 'https://www.comicrelief.com/news',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://www.comicrelief.com/press-releases',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998a',
      title: 'Legal',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Privacy policy',
          url: 'https://www.comicrelief.com/privacy-notice',
          class: null
        },
        {
          title: 'Terms of use',
          url: 'https://www.comicrelief.com/terms-of-use',
          class: null
        },
        {
          title: 'Cookies',
          url: 'https://www.comicrelief.com/cookies-policy',
          class: null
        },
        {
          title: 'Positive practices',
          url: 'https://www.comicrelief.com/positive-practices',
          class: null
        },
        {
          title: 'Text to donate terms of use',
          url: 'https://www.comicrelief.com/legal/sms',
          class: null
        }
      ]
    }
  ]
};
