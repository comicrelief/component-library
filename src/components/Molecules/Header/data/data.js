export default {
  menuDescription: 'I am the navigation',
  menuGroup: [
    {
      id: 'Parent Item Title 1',
      title: 'Parent Item Title 1',
      url: '/parent1',
      links: [
        {
          title: 'Child Item Title 1-1',
          url: '/child-item-1-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-2',
          url: '/child-item-1-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-3',
          url: '/child-item-1-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 2',
      title: 'Parent Item Title 2',
      url: '/parent2',
      links: [
        {
          title: 'Child Item Title 2-1',
          url: '/child-item-2-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-2',
          url: '/child-item-2-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-3',
          url: '/child-item-2-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 4',
      title: 'Parent Item Title 4',
      url: '/parent4',
      links: [
        {
          title: 'Child Item Title 4-1',
          url: '/child-item-4-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-4',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    }
  ]
};
