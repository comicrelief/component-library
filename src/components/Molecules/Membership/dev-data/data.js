export default {
  cartId: 'test-sept2019-C365',
  regularGiving: {
    otherDescription:
      'Your donation will help us fund amazing projects in the UK and around the world.',
    moneybuys: [
      {
        amount: 10,
        description:
          'a regular supply of toiletries for someone living in a refugee camp in Serbia.'
      },
      {
        amount: 20,
        description:
          'a potentially lifesaving call for a man at risk of suicide in the UK.'
      },
      {
        amount: 30,
        description:
          'the distribution of enough surplus food for a school breakfast club to feed 80 children.'
      }
    ]
  }
};
