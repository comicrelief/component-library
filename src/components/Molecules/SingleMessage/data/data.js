export default {
  title: 'What your money does',
  text:
    'When you donate to Comic Relief or Sport Relief, youâ€™re supporting vulnerable people and communities in the UK and internationally.',
  pictures: {
    small: 'src/components/Atoms/Picture/assets/picture-1.jpg',
    medium: 'src/components/Atoms/Picture/assets/picture-2.jpg',
    large: 'src/components/Atoms/Picture/assets/picture-3.jpg'
  },
  ctaText: 'Find out how'
};
