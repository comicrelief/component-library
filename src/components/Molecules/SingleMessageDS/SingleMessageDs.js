import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import Picture from '../../Atoms/Picture/Picture';
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import { External, Internal } from '../../Atoms/Icons/index';
import spacing from '../../../theme/shared/spacing';
import zIndex from '../../../theme/shared/zIndex';

const Container = styled.div`
  display: flex;
  position: relative;
  flex-direction: column;
  height: 100%;
  width: 100%;
  background: ${({ theme, backgroundColor }) => theme.color(backgroundColor)};

  @media ${({ theme }) => theme.breakpoint('small')} {
    flex-direction: ${({ direction }) => (direction === 'right' ? 'row-reverse' : 'row')};
    width: 100%;
  }
`;

const Heading = styled(Text)`
  letter-spacing: 0.03em;
  text-transform: uppercase;
`;

const Title = styled(Heading)`
  margin: ${spacing('md')} 0;
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('m')};
  }
`;

const Subtitle = styled(Heading)`
  display: inline-block;
`;

const Image = styled.div`
  height: auto;
  margin: 0 ${spacing('m')} 0 -${spacing('m')};
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin: 0;
    ${({ hasLink }) => !hasLink && `width: calc(50% - ${spacing('xl')})`};
  }

  @media ${({ theme }) => theme.breakpoint('large')} {

    margin: 0 -${spacing('m')} 0 ${spacing('m')};
  }
  img {
    border-radius: 1rem;
    box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.15);
  }
`;

const MediaLink = styled.a`
  width: 320px;
  @media ${({ theme }) => theme.breakpoint('small')} {
    width: 360px;
  }

  @media ${({ theme }) => theme.breakpoint('large')} {
    width: 600px;
  }
`;

const Copy = styled.div`
  position: relative;
  width: 320px;
  padding: ${spacing('m')};
  ${zIndex('low')};
  display: flex;
  flex-direction: column;
  border-radius: 1rem;
  box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.15);
  background: ${({ theme, backgroundColor }) => theme.color(backgroundColor)};

  ${({ hasLink }) => hasLink && `padding-bottom: calc(${spacing('l')} + ${spacing('m')})`};

  ${({ hasImage }) => hasImage
    && css`
      min-height: calc(5 * ${spacing('l')});
      @media ${({ theme }) => theme.breakpoint('small')} {
        margin: ${({ direction }) => (direction === 'left'
    ? `${spacing('l')} 0 -${spacing('m')} calc(-${spacing('m')} * 2)`
    : `${spacing('l')} calc(-${spacing('m')} * 2) -${spacing('m')} 0`)};
    }
    `};

  @media ${({ theme }) => theme.breakpoint('small')} {
    width: 360px;
    padding: ${({ direction }) => (direction === 'left'
    ? `${spacing('l')} ${spacing('m')} ${spacing('l')} ${spacing('l')}`
    : `${spacing('l')} ${spacing('l')} ${spacing('l')} ${spacing('m')}`)};
    ${({ hasLink }) => hasLink && `padding-bottom: calc(${spacing('l')} + ${spacing('m')})`};
  }

  @media ${({ theme }) => theme.breakpoint('large')} {
    width: 600px;
    background: red;
    height: 100%;
    padding: calc(${spacing('m')} * 2);
    ${({ hasLink }) => hasLink && `padding-bottom: calc(${spacing('m')} * 3)`};

    ${({ hasImage }) => hasImage
    && css`
      min-height: calc(5 * ${spacing('l')});
      @media ${({ theme }) => theme.breakpoint('small')} {
        margin: ${({ direction }) => (direction === 'left'
    ? `${spacing('l')} 0 ${spacing('m')} -${spacing('l')}`
    : `${spacing('l')} calc(-${spacing('l')} * 2) ${spacing('m')} 0`)};
    }
    `};
  }
`;

const CTA = styled.div`
  position: absolute;
  right: ${spacing('m')};
  bottom: -${spacing('m')};
  ${zIndex('medium')};

  a {
    font-family: ${({ theme }) => theme.fontFamilies('Montserrat')};
  }

  @media ${({ theme }) => theme.breakpoint('small')} {
    ${({ hasImage }) => !hasImage && `bottom: -${spacing('m')};`}
  }
`;

const SingleMessageDs = ({
  backgroundColor,
  imageDirection,
  imageLow,
  images,
  image,
  imageAltText,
  height,
  width,
  subtitle,
  title,
  children,
  link,
  linkLabel,
  ctaBgColor,
  target,
  ...rest
}) => {
  const Media = (
    <Image hasLink={link}>
      <Picture
        alt={imageAltText}
        imageLow={imageLow}
        images={images}
        image={image}
        objectFit="cover"
        width={width}
        height={height}
      />
    </Image>
  );

  const hasMedia = () => {
    if (imageLow && link) {
      return (
        <MediaLink
          aria-hidden="true"
          tabIndex="-1"
          href={link}
          target={target}
          {...rest}
        >
          {Media}
        </MediaLink>
      );
    }
    if (imageLow) {
      return Media;
    }
    return null;
  };

  const icon = () => {
    if (target === 'blank') {
      return <External colour="white" />;
    }
    return <Internal colour="white" />;
  };

  const external = target === 'blank' ? 'noopener noreferrer' : null;

  return (
    <Container {...rest} direction={imageDirection}>
      {hasMedia()}
      <Copy
        hasImage={imageLow}
        hasLink={link}
        backgroundColor={backgroundColor}
        direction={imageDirection}
      >
        <Subtitle color="blue_dark" size="s" weight="bold" family="Montserrat">
          {subtitle}
        </Subtitle>
        <Title tag="h4" color="grey_dark" size="xl" family="Anton" weight="normal">
          {title}
        </Title>
        {children}
        {link && (
        <CTA hasImage={imageLow}>
          <Link
            rel={external}
            color={ctaBgColor}
            href={link}
            target={target}
            type="button"
            icon={icon()}
          >
            {linkLabel}
          </Link>
        </CTA>
        )}
      </Copy>
    </Container>
  );
};

SingleMessageDs.propTypes = {
  backgroundColor: PropTypes.string,
  imageDirection: PropTypes.string,
  imageLow: PropTypes.string,
  images: PropTypes.string,
  image: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  imageAltText: PropTypes.string,
  link: PropTypes.string,
  ctaBgColor: PropTypes.string,
  linkLabel: PropTypes.string,
  subtitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  target: PropTypes.string,
  children: PropTypes.node.isRequired
};

SingleMessageDs.defaultProps = {
  backgroundColor: 'white',
  imageDirection: 'left',
  imageLow: null,
  images: null,
  image: null,
  link: null,
  ctaBgColor: 'red',
  linkLabel: null,
  target: null,
  imageAltText: '',
  width: '100%',
  height: '100%'
};

export default SingleMessageDs;
