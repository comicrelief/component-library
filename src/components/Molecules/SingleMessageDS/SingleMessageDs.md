# SingleMessageDs

## Image align left and CTA are clickable and links user through to full conten

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';

const Title = styled(Text)`
  letter-spacing: 0.03em;
  text-transform: uppercase;
  margin: ${spacing('md')} 0;
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('m')};
  }
`;
<div
    style={{
      display: 'flex',
      padding: '4rem 0',
      background: '#E1E2E3',
      justifyContent: 'space-around',
    }}>
    <SingleMessageDs
      imageLeft={true}
      link='/home'
      linkLabel='Check out'
      backgroundColor='white'
      imageLow={data.image}
      images={data.images}
      imageAltText='Happy man going to work'
      subtitle="Subtitle"
      height='100%'
      hasVideo={true}
    >
      <Title tag="h4" color="grey_dark" size="xl" family="Anton" weight="normal">
        Heading Line 1 Heading Line 2
      </Title>
      <Text tag='p'>
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
</div>;
```

## Image align right  and CTA are clickable and links user through to full content

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';

const Title = styled(Text)`
  margin: ${spacing('md')} 0;
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('m')};
  }
`;
<div
    style={{
      display: 'flex',
      padding: '4rem 0',
      background: '#E1E2E3',
      justifyContent: 'space-around',
    }}>
    <SingleMessageDs
      target='blank'
      link='https://www.comicrelief.com/'
      linkLabel='Check out'
      ctaBgColor="blue_dark"
      backgroundColor='white'
      imageLeft={false}
      imageLow={data.image}
      images={data.images}
      imageAltText='Happy man going to work'
      subtitle="Subtitle"
      height='100%'
    >
      <Title tag="h4" color="grey_dark" size="xl" family="Anton" weight="normal">
        Heading Line 1 Heading Line 2
      </Title>
      <Text tag='p'>
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
</div>;
```

## Image align with non CTA

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';

const Title = styled(Text)`
  margin: ${spacing('md')} 0;
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('m')};
  }
`;
<div
    style={{
      display: 'flex',
      padding: '4rem 0',
      background: '#E1E2E3',
      justifyContent: 'space-around',
    }}>
    <SingleMessageDs
      linkLabel='Check out'
      ctaBgColor="blue_dark"
      backgroundColor='white'
      imageLeft={true}
      imageLow={data.image}
      images={data.images}
      imageAltText='Happy man going to work'
      subtitle="Subtitle"
      height='100%'
    >
      <Title tag="h4" color="grey_dark" size="xl" family="Anton" weight="normal">
        Heading Line 1 Heading Line 2
      </Title>
      <Text tag='p'>
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
</div>;
```

## Using the linkIcon prop (to display download icon)

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';
import Download from '../../Atoms/Icons/Download';

<div>
    <SingleMessageDs
      link="https://assets.ctfassets.net/zsfivwzfgl3t/EsIEuI1teYkbLB7peOtqP/b43a905cffd89115d642fe5a84ff020a/RND21_fundraise_moneybuys_poster.pdf"
      linkLabel='Download this'
      linkIcon={<Download />}
      imageLow={data.image}
      images={data.images}
    >
      <Text tag='p'>
        Some text and stuff.
      </Text>
    </SingleMessageDs>
</div>;
```
