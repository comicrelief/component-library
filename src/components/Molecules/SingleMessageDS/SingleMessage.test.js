import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import SingleMessageDs from './SingleMessageDs';
import Text from '../../Atoms/Text/Text';
import data from '../../../styleguide/data/data';

it('renders correctly', () => {
  const tree = renderWithTheme(
    <SingleMessageDs
      target="_blank"
      link="/home"
      linkLabel="Check out the shop"
      ctaBgColor="blue_dark"
      backgroundColor="white"
      imageDirection="right"
      imageLow={data.image}
      images={data.images}
      imageAltText="Happy man going to work"
      subtitle="Subtitle"
      title="Heading Line 1 Heading Line 2"
      height="auto"
    >
      <Text tag="p">
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
