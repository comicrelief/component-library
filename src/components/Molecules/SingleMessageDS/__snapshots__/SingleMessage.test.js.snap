// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c6 {
  color: #274084;
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  font-weight: bold;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c8 {
  color: #222222;
  font-size: 2rem;
  line-height: 2rem;
  text-transform: inherit;
  font-weight: normal;
  font-family: 'Anton',Impact,sans-serif;
  -webkit-letter-spacing: 0.03rem;
  -moz-letter-spacing: 0.03rem;
  -ms-letter-spacing: 0.03rem;
  letter-spacing: 0.03rem;
}

.c10 {
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c3 {
  position: relative;
  display: block;
  width: 100%;
  height: auto;
}

.c4 {
  display: block;
  width: 100%;
  height: auto;
  object-fit: cover;
}

.c13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: 1rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  right: 1.5rem;
}

.c12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  padding: 0.5rem 1.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  font-size: 1rem;
  border-radius: 2rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  height: 3.125rem;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background-color: #274084;
  color: #FFFFFF;
}

.c12:hover {
  background-color: #0565D1;
  color: #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  background: inherit;
}

.c9 {
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  text-transform: uppercase;
  margin: 1rem 0;
}

.c7 {
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  text-transform: uppercase;
  display: inline-block;
}

.c2 {
  height: auto;
  margin: 0 1.5rem 0 -1.5rem;
}

.c2 img {
  border-radius: 1rem;
  box-shadow: 0px 0px 20px rgba(0,0,0,0.15);
}

.c1 {
  width: 320px;
}

.c5 {
  position: relative;
  width: 320px;
  padding: 1.5rem;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 1rem;
  box-shadow: 0px 0px 20px rgba(0,0,0,0.15);
  background: #FFFFFF;
  padding-bottom: calc(2rem + 1.5rem);
  min-height: calc(5 * 2rem);
}

.c11 {
  position: absolute;
  right: 1.5rem;
  bottom: -1.5rem;
  z-index: 2;
}

.c11 a {
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

@media (min-width:740px) {
  .c13 {
    width: auto;
    right: auto;
    position: relative;
  }
}

@media (min-width:740px) {
  .c12 {
    width: auto;
  }
}

@media (min-width:740px) {
  .c0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    width: 100%;
  }
}

@media (min-width:740px) {
  .c9 {
    margin-bottom: 1.5rem;
  }
}

@media (min-width:740px) {
  .c2 {
    margin: 0;
  }
}

@media (min-width:1440px) {
  .c2 {
    margin: 0 -1.5rem 0 1.5rem;
  }
}

@media (min-width:740px) {
  .c1 {
    width: 360px;
  }
}

@media (min-width:1440px) {
  .c1 {
    width: 600px;
  }
}

@media (min-width:740px) {
  .c5 {
    margin: 2rem calc(-1.5rem * 2) -1.5rem 0;
  }
}

@media (min-width:740px) {
  .c5 {
    width: 360px;
    padding: 2rem 2rem 2rem 1.5rem;
    padding-bottom: calc(2rem + 1.5rem);
  }
}

@media (min-width:1440px) {
  .c5 {
    width: 600px;
    background: red;
    height: 100%;
    padding: calc(1.5rem * 2);
    padding-bottom: calc(1.5rem * 3);
    min-height: calc(5 * 2rem);
  }
}

@media (min-width:740px) {

}

<div
  className="c0"
  direction="right"
>
  <a
    aria-hidden="true"
    className="c1"
    href="/home"
    tabIndex="-1"
    target="_blank"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        height="auto"
        width="100%"
      >
        <img
          alt="Happy man going to work"
          className="c4 lazyload"
          data-lowsrc="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
          data-sizes="auto"
          data-src={null}
          data-srcset="//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=200&h=150&q=50 200w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=400&h=300&q=50 400w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=800&h=600&q=50 800w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=1200&h=900&q=50 1200w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=1440&h=1080&q=50 1440w"
          height="auto"
          src={null}
          srcSet="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          width="100%"
        />
      </div>
    </div>
  </a>
  <div
    className="c5"
    direction="right"
  >
    <span
      className="c6 SingleMessageDs__Heading-sc-1dprx4j-1 c7"
      color="blue_dark"
      size="s"
    >
      Subtitle
    </span>
    <h4
      className="c8 SingleMessageDs__Heading-sc-1dprx4j-1 c9"
      color="grey_dark"
      size="xl"
    >
      Heading Line 1 Heading Line 2
    </h4>
    <p
      className="c10"
      color="inherit"
      size="s"
    >
      Whatever you’ve got planned, the Sport Relief shop has everything you need to get you looking your best while you’re raising some cash. Also available in Sainsbury’s stores and online and in selected Argos stores.
    </p>
    <div
      className="c11"
    >
      <a
        className="c12"
        color="blue_dark"
        href="/home"
        rel={null}
        target="_self"
        type="button"
      >
        Check out the shop
        <span
          className="c13"
          type="button"
        >
          <svg
            fill="#FFFFFF"
            heiht={24}
            viewBox="0 0 96 96"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M90.871 50.776c-.016.016-.026.036-.043.052l-32 32C58.047 83.609 57.024 84 56 84s-2.047-.391-2.829-1.171a4 4 0 010-5.657L78.343 52H8a4 4 0 010-8h70.343L53.171 18.829a4 4 0 015.657-5.657l32 32c.016.016.027.036.043.052.167.172.321.357.455.557.069.104.118.216.176.325.065.121.138.236.191.364.057.137.09.279.13.42.032.109.074.214.097.327a4.017 4.017 0 010 1.568c-.023.113-.065.217-.097.327-.041.141-.074.283-.13.42-.053.128-.126.243-.191.364-.059.109-.107.221-.176.325a4.035 4.035 0 01-.455.555z"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
</div>
`;
